src/components/SearchManager/SearchManager.vue

```diff
@@ -1,61 +1,53 @@
 <template>
-    <div class="search_manager">
-        <span class="search_manager__count">
-            {{ count }} movies found
-        </span>
-        <span class="search_manager__sort-by">
-            <span class="search_manager__sort-by-text">Sort by</span>
-            <ButtonsBar v-bind:buttons="buttons" v-bind:onButtonClick="handleSortChange" v-bind:selectedId="currentSortId" />
-        </span>
-    </div>
+  <div class="search_manager">
+    <span class="search_manager__count"> {{ count }} movies found </span>
+    <span class="search_manager__sort-by">
+      <span class="search_manager__sort-by-text">Sort by</span>
+      <ButtonsBar
+        :buttons="sortOptions"
+        :selectedId="sortById"
+        v-on:buttonClick="$emit('sortChange', $event)"
+      />
+    </span>
+  </div>
 </template>
 
 <script>
-  import ButtonsBar from "../ButtonsBar/ButtonsBar";
+import ButtonsBar from "../ButtonsBar/ButtonsBar";
 
-  export default {
-    components: {ButtonsBar},
-    name: "SearchManager",
-    props: {
-      count: Number
-    },
-    data: () => ({
-      buttons: [{
-        id: 'release_date',
-        title: 'release date'
-      }, {
-        id: 'rating',
-        title: 'rating'
-      }],
-      currentSortId: 'release_date'
-    }),
-    methods: {
-      handleSortChange: function(newSortParameter) {
-        this.currentSortId = newSortParameter;
-      }
+export default {
+  components: { ButtonsBar },
+  name: "SearchManager",
+  props: {
+    count: Number,
+    sortOptions: Array,
+    sortById: String
+  },
+  methods: {
+    handleSortChange: function(newSortParameter) {
+      this.$emit("sortChange", newSortParameter);
```

Comment:
👍  good attempt, but you don't call this method, seems you emit event in the template